### ISSUE TITLE
[1.0.0-rc.1] [Windows] Error setting url to '...' (local file)

### ISSUE THREAD

**Issue by cfuj** [bluefireteam/audioplayers#1100]
`AudioPlayer.play(DeviceFileSource(file.path))` cannot get local file on windows. It says "Error setting url"

**Full Description**
I'm using `flutter_cache_manager` in conjunction with this addon to download a series of audio files from my server. This works fine on Android, as yet untested on iOS.

On Windows, it gives me Error setting url to Error setting url to 'C:\Users\MyUser\AppData\Local\Temp\MyApp\audio\a3f1cfd0-bb89-11ec-a8bd-1d244b97d5ad.file'

I've also tried using a BytesSource instead, but that reports an unimplemented error for Windows.

**Code to Reproduce**

    var MyAudioCache = CacheManager(
      Config(
        "audio", // cache name
        stalePeriod: Duration(days: 1),
        repo: JsonCacheInfoRepository(databaseName: "audio"),
        fileService: HttpFileService(),
      ),
    );

    var url = "some_url";

    await MyAudioCache.getSingleFile(url, key: url);
    var file = (await MyAudioCache.getFileFromCache(url))!.file;
    var ap = AudioPlayer();
    ap.play(DeviceFileSource(file.path));

I know the problem is here because I'm able to use an alternative, but very rough plugin, `kPlayer' and use

    kPlayer = Player.asset(file.path, autoPlay: true);

**Log Errors**
AudioPlayers: Error setting url to 'C:\Users\MyUser\AppData\Local\Temp\MyApp\audio\a3f1cfd0-bb89-11ec-a8bd-1d244b97d5ad.file'.

**Files/URLs/Sources**

**Screenshots**
If applicable, add screenshots or video recordings to help explain your problem. This is totally optional.

**Platforms**
Android 8.1: Works
Windows 11: Doesn't work

* Device: My main System, main OS (not a VM), custom built
* flutter version: 2.10.4
* audioplayers version: 1.0.0-rc.1
* release or not release: debug mode

**Comment by shirenhu** [comment]
some Error

**Comment by luojt** [comment]
me too

**Comment by merabtenei** [comment]
I can only play .wav files on windows, no support for m4a, ogg etc...

**Comment by Gustl22** [comment]
@merabtenei can you provide the files?

**Comment by merabtenei** [comment]
> @merabtenei can you provide the files?

Thank for taking the time to look at this.
I use the record library in pub dev to record a file and play it with audioplayers.
I prepared a complete example to reproduce the issue :
```
import 'dart:io';

import 'package:audioplayers/audioplayers.dart';
import 'package:flutter/material.dart';
import 'package:path_provider/path_provider.dart';
import 'package:record/record.dart';
import 'package:path/path.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Record Play audio',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: const MyHomePage(title: 'Demo record/play audio'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key, required this.title});

  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  final record = Record();
  bool isRecording = false;

  Future<void> recordAndPlay(Duration duration) async {
    if (await record.hasPermission()) {
      Directory? dir = await getDownloadsDirectory();
      if (dir == null) return;
      await record.start(
        path: join(dir.path, 'test'),
        encoder: AudioEncoder.aacLc, // by default
        bitRate: 128000, // by default
        samplingRate: 44100, // by default
      );
      setState(() {
        isRecording = true;
      });
      await Future.delayed(duration);
      String? filePath = await record.stop();
      setState(() {
        isRecording = false;
      });
      if (filePath != null) {
        final player = AudioPlayer();
        await player.play(DeviceFileSource(filePath));
      }
    }
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
      ),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            ElevatedButton(
              onPressed: () async {
                await recordAndPlay(const Duration(seconds: 8));
                print('DONE');
              },
              child: const Text('Record & Play'),
            ),
            if (isRecording) const Text('Recording ...')
          ],
        ),
      ),
    );
  }
}
```
Run this on windows and click the record and play button. This will record an 8 seconds test.m4a file and save it to Downloads folder. When trying to play that file I get : ```AudioPlayers: Error setting url to 'D:\Downloads\test.m4a'.```

Need to add these to pubspec.yaml:
```
  record: ^4.4.3
  audioplayers: ^1.1.1
  path_provider: ^2.0.11
```

**Comment by Gustl22** [comment]
I'm sorry for the delay, there are many bugs, and I do this for free, so unfortunately I have not the time to do the recording. Can you upload one of the `.m4a` recordings (wrap in  `.zip` if github doesn't like it) ? And even better, can you test if you can play the file with our example app?
[This site](https://www.aconvert.com/analyze.html) provides very precise information about the format, and then can see if it's compatible to the supported ones of [windows 11](https://learn.microsoft.com/en-us/windows/win32/medfound/supported-media-formats-in-media-foundation)

I have the feeling that the `.m4a` container is containing some old codec this isn't supported by windows anymore, such as `.mp2`. You should contact the maintainer of the `record` package, to support also e.g. `mp3` which is supported by their used package: [fmedia](https://stsaz.github.io/fmedia/)
Or do a PR in record package: https://github.com/llfbandit/record/pulls

**Comment by Gustl22** [comment]
I closed it, as I expect this to be related to #1494. Feel free to open again or raise a new issue, if it is not the case. Thanks for understanding.
